src/VocabularyManager.UseCases/DTOs/VocabularyDTO.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace VocabularyManager.UseCases.DTOs
+{
+    public class VocabularyDTO
+    {
+        public int? Id { get; init; } = 0;
+        public string Name { get; set; }
+        public string? SourceUrl { get; set; }
+        public List<WordDTO>? Words { get; set; } = new List<WordDTO>();
```

Comment:
the list won't ever be `?` as you are initializing it so I think you can remove `?`

Same for above on int id